views/bigquery.go

```diff
@@ -155,18 +155,18 @@ func (bq *bigqueryHandler) QueryViewsEvents(ctx context.Context, spec QuerySpec)
 func buildViewsEventsQuery(table string, spec QuerySpec) (string, []interface{}, error) {
 	query := squirrel.Select(
 		"countif(play_intent) as view_count",
-		"sum(playtime_ms) / 60000.0 as playtime_mins").
+		"ifnull(sum(playtime_ms), 0) / 60000.0 as playtime_mins").
 		From(table).
 		Where("account_id = ?", spec.Filter.UserID).
 		Limit(maxBigQueryResultRows + 1)
 	query = withPlaybackIdFilter(query, spec.Filter.PlaybackID)
 
 	if spec.Detailed {
 		query = query.Columns(
-			"avg(ttff_ms) as ttff_ms",
-			"avg(rebuffer_ratio) as rebuffer_ratio",
-			"avg(if(error_count > 0, 1, 0)) as error_rate",
-			"sum(if(exit_before_start, 1.0, 0.0)) as exits_before_start")
+			"ifnull(avg(ttff_ms), 0) as ttff_ms",
+			"ifnull(avg(rebuffer_ratio), 0) as rebuffer_ratio",
+			"ifnull(avg(if(error_count > 0, 1, 0)), 0) as error_rate",
+			"ifnull(sum(if(exit_before_start, 1.0, 0.0)), 0) as exits_before_start")
```

Comment:
These metrics _are_ optional, and the 0 value would have a different semantic here (e.g. 0ms ttff), so I believe they should be kept nulable. Or at least that's how it's currently implemented, with the nulls being explicitly handled with the nullables etc, so I think we should keep as is.

Reply:
Ok, I've kept them as null